tap_postgres/tests/test_core.py

```diff
@@ -1,87 +1,163 @@
-from faker import Faker
-import pendulum
-import datetime
-import pytest
-import json
-from sqlalchemy import Table, Column, String, DateTime, Integer, MetaData
 
-SAMPLE_CONFIG = {
-    "start_date": pendulum.datetime(2022,11,1).to_iso8601_string(), 
-    "sqlalchemy_url": "postgresql://postgres:postgres@localhost:5432/postgres",
-}
 
+@pytest.fixture
+def sample_config_components(postgres_fixture):
+    """A config that uses the user/password/host/port settings to configure
+    database connection.
+    """
+    postgres_host_pytest_docker = postgres_fixture.get_addr("5432/tcp")[0]
+    return {
+        "start_date": pendulum.datetime(2022, 11, 1).to_iso8601_string(),
+        "host": os.getenv("TAP_POSTGRES_HOST", postgres_host_pytest_docker),
+        "port": os.getenv("TAP_POSTGRES_PORT", 5432),
+        "user": os.getenv("TAP_POSTGRES_USER", "postgres"),
+        "password": os.getenv("TAP_POSTGRES_PASSWORD", "mypass"),
+        "database": os.getenv("TAP_POSTGRES_DATABASE", "postgres"),
+    }
+
+
+@pytest.fixture
+def sample_config_sqlalchemy_url(postgres_fixture):
+    """A config that uses the sqlalchemy_url setting to configure
+    database connection.
+    """
+    postgres_host_pytest_docker = postgres_fixture.get_addr("5432/tcp")[0]
+    return {
+        "start_date": pendulum.datetime(2022, 11, 1).to_iso8601_string(),
+        "sqlalchemy_url": os.getenv(
+            "TAP_POSTGRES_SQLALCHEMY_URL",
+            (
+                "postgresql+psycopg2://postgres:mypass@"
+                f"{postgres_host_pytest_docker}:5432/postgres"
+            ),
+        ),
+    }
 
-# Run standard built-in tap tests from the SDK:
-def test_standard_tap_tests():
-    """Run standard tap tests from the SDK."""
-    tests = get_standard_tap_tests(
-        TapPostgres,
-        config=SAMPLE_CONFIG
-    )
-    for test in tests:
-        test()
 
 @pytest.fixture()
 def sqlalchemy_connection(tap):
-    #Used this function to avoid server side cursors which don't work with DDL
-    return tap.default_stream_class.connector_class(dict(tap.config)).create_sqlalchemy_engine().connect()
+    # Used this function to avoid server side cursors which don't work with DDL
+    return (
+        tap.default_stream_class.connector_class(dict(tap.config))
+        .create_sqlalchemy_engine()
+        .connect()
+    )
+
 
 @pytest.fixture()
-def tap():
-    return TapPostgres(config=SAMPLE_CONFIG)
+def tap(sample_config_components):
+    sample_config = sample_config_components
+    return TapPostgres(config=sample_config)
+
 
 @pytest.fixture()
 def fake():
     return Faker()
 
-def test_replication_key(sqlalchemy_connection, tap:TapPostgres, fake):
-    """Originally built to address https://github.com/MeltanoLabs/tap-postgres/issues/9"""
-    date1 = datetime.date(2022,11,1)
-    date2 = datetime.date(2022,11,30)
+
+# Run standard built-in tap tests from the SDK:
+def test_standard_tap_tests(sample_config_components):
```

Comment:
Awesome, do these pass? If not please create an issue for us to dive into this.

Reply:
Yep, they pass! The other change in this PR to make the tap use a singleton connector solves the too many clients error (#6)